rust/src/meas/randomized_response/mod.tex

```diff
@@ -0,0 +1,250 @@
+    %\item \textbf{(Appropriate output domain.)} In the case of \texttt{binary\_randomized\_response}, this corresponds to showing that for every element $v$ in \texttt{AllDomain(bool)}, the element \texttt{function(v)} belongs to \texttt{AllDomain(bool)}. The type signature of \texttt{function} as defined in line \ref{line:fn} automatically enforces that \texttt{function(v)} has type \texttt{bool}. Since the Rust code successfully compiles, by the type signature the appropriate output domain property must hold. Otherwise, the code will raise an exception for incorrect input type. 
+    
+    %Another way to see this is that \texttt{function(v)} returns either \emph{arg} or \emph{lie}. From the function signature, \emph{arg} is of type \texttt{bool}. As shown above, \texttt{sample\_bernoulli} can either return 0 or 1, so it is therefore also restricted to boolean values. Hence, \emph{lie} must also be boolean. 
+    
+    %Therefore, \texttt{function(v)} is guaranteed to be in \texttt{output\_domain} in all cases.
+    
+    % need to check for overflow? 
+
+    \item \textbf{(Domain-metric compatibility.)} For \texttt{binary\_randomized\_response}, this corresponds to showing \texttt{AllDomain(bool)} is compatible with \texttt{DiscreteMetric}. This follows directly from the definition of \texttt{DiscreteMetric}, as stated in the \href{https://github.com/opendp/whitepapers/blob/pseudocode-defns/pseudocode-defns/pseudocode_defns.pdf}{``List of definitions used in the pseudocode"}.
+    
+    \item \textbf{(Privacy guarantee.)} 
+    % We assume proof of correctness of the \texttt{bernoulli\_sampler} as shown in the document ??? (to do). 
+    
+    \begin{tcolorbox}
+\begin{note}[Proof relies on correctness of Bernoulli sampler]
+The following proof makes use of the following lemma that asserts the correctness of the Bernoulli sampler function.
+    \begin{lemma}
+    \texttt{sample\_bernoulli(prob, constant\_time)}, the Bernoulli sampler function used in \texttt{make\_randomized\_response\_bool}, returns \texttt{true} with probability (\texttt{prob}) and returns  \texttt{false} with probability (1 - \texttt{prob}).
+    \end{lemma}
+\end{note}
+\end{tcolorbox}
+    \vicki{to do: need to relax the epsilon-delta defns.} 
+    
+    Let $v$ and $w$ be datasets that are \texttt{d\_in}-close with respect to \texttt{input\_metric}. Here, the metric is \texttt{DiscreteMetric} which enforces that $\din = 1$ if $v \ne w$ and $\din = 0$ if $v = w$. The case where $v = w$ is trivial so we only consider $v \ne w$ and assume without loss of generality that $v = \texttt{true}$ and $w = \texttt{false}$. For shorthand, we let $p$ represent \texttt{prob}, the probability that \texttt{sample\_bernoulli} returns \texttt{true}. Observe that $p = [0.5, 1.0)$ otherwise \texttt{make\_randomized\_response\_bool} raises an error. 
+    
+    We now consider the max-divergence $D_{\infty}(Y||Z)$ over the random variables $Y = \function(v)$ and $Z = \function(w)$.
+    
```

Comment:
Throughout the sequence of equations below, use \left and \right for readability.